lib/puppet/type/wait_for.rb

```diff
@@ -160,28 +187,55 @@ def check_all_attributes(refreshing = false)
 
     validate do |value|
       raise ArgumentError,
-        "Regex must be a String, got value of class #{value.class}" unless value.is_a?(String)
+        "Regex must be of type String, got value of class #{value.class}" unless value.is_a?(String)
     end
   end
 
-  newproperty(:seconds) do
+  newproperty(:state) do |property|
     include Mixins
 
+    desc "If path is specified: Whether the filesystem element should be absent,
+      present, a file or a directory.
+
+      When present is specified, we only test for existance, regardles of type."
+
+    newvalues(:absent, :directory, :file, :present)
+  end
+
+  newproperty(:seconds) do
+    include Mixins
     desc "Just wait this number of seconds no matter what."
+
     munge do |value|
-      value.to_i
+      value.to_f
+    end
+  end
+
+  newparam(:title, :namevar => true) do
+    desc "A short, unique description of what we're waiting for."
+
+    validate do |value|
+      raise ArgumentError,
+        "Title must be of type String, got value of class #{value.class}" unless value.is_a?(String)
+    end
+  end
+
+  newparam(:path) do
+    desc "A path on the filesystem which should (dis-)appear."
+
+    validate do |value|
+      raise ArgumentError,
+        "Path must be of type String, got value of class #{value.class}" unless value.is_a?(String)
     end
   end
 
   newparam(:query) do
     desc "The command to execute. The output of this command
       will be matched against the regex."
 
-    isnamevar
```

Comment:
Is removing this intentional?

Reply:
Subsequent to this, I noted your comment above about needing to not have a namevar. If you're certain there is no way to keep namevar then so be it. I couldn't quite get my head around why that would be however.

Reply:
The comment is not about having no namevar (in fact, you can't), but having a different one (`title`). Using `query` as namevar doesn't make much sense when we can operate in three different modes. We should have done this already when we introduced `seconds`.

Reply:
Fair enough @heini , I think you mean title would be the namevar in the sense that any resource can be referred to by title? I don't see any real issue leaving namevar on query. The advantage would be that you don't break backwards compatibility and the type remains more convenient for those who only want the query feature. I guess you are feeling that it is a bit inconsistent that one feature gets namevar and none of the others do? It's up to you. It's no big deal for me to refactor for my use-case if the namevar changes.